apps/web/src/views/Home/hooks/useFarmsWithBalance.tsx

```diff
@@ -31,13 +34,28 @@ const useFarmsWithBalance = () => {
     accountToCheck: string,
     contract: Masterchef | BCakeProxy,
   ) => {
-    const calls = farms.map((farm) => ({
+    const masterChefCalls = farms.map((farm) => ({
+      abi: masterChefABI,
       address: masterChefContract.address,
       name: 'pendingCake',
       params: [farm.pid, accountToCheck],
     }))
 
-    const rawResults = await multicall(masterChefABI, calls)
+    const proxyCall =
+      contract.address !== masterChefContract.address && bCakeProxy
+        ? {
+            abi: cakeAbi,
+            address: CAKE[chainId].address,
+            name: 'balanceOf',
+            params: [bCakeProxy.address],
+          }
+        : null
+
+    const calls = [...masterChefCalls, proxyCall].filter(Boolean)
+
+    const rawResults = await multicallv3({ calls })
+    const proxyCakeBalance = proxyCall ? rawResults.pop() : null
```

Comment:
should check `rawResults` length before calling `pop`. It potentially throws exception